backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Docs_Sales_Invoice_QR_Code.sql

```diff
@@ -37,107 +38,103 @@ select ('SPC' || E'\n' || --QRType
-        coalesce(l.postal, '') || ' ' || coalesce(l.city, '') || E'\n' || -- UD Postal code and town
+        SUBSTRING(bpartneraddress FROM 1 FOR POSITION(E'\n' IN bpartneraddress)) || --UD – Name
+        COALESCE(l.address1, '') || E'\n' || --UD –Street and building number of P.O. Box
+        COALESCE(l.postal, '') || ' ' || COALESCE(l.city, '') || E'\n' || -- UD Postal code and town
         E'\n' || --Do not fill in
         E'\n' || --Do not fill in
         c.countrycode || E'\n' || -- UD Country
-        (case
-             when replace(coalesce(orgbpb_invoice.qr_iban, orgbpb.iban,''), ' ', '') is not null and rn.referenceNo is not null then 'QRR'
-             when replace(coalesce(orgbpb_invoice.iban, orgbpb.iban,''), ' ', '') is not null and replace(orgbpb.sepa_creditoridentifier, ' ', '') is not null
-                 then 'SCOR'
-             else 'NON'
-            end) || E'\n' || --ReferenceType
-        (case
-             when replace(coalesce(orgbpb_invoice.qr_iban, orgbpb.iban,''), ' ', '') is not null and rn.referenceNo is not null then rn.ReferenceNo
-             when replace(coalesce(orgbpb_invoice.iban, orgbpb.iban,''), ' ', '') is not null and replace(orgbpb.sepa_creditoridentifier, ' ', '') is not null
-                 then 'RF' || orgbpb.sepa_creditoridentifier
-             else ''
-            end) || E'\n' || --Reference
-        coalesce(i.description,'') || E'\n' ||--Unstructured message
+        (CASE
+             WHEN REPLACE(COALESCE(orgbpb_invoice.qr_iban, orgbpb.iban, ''), ' ', '') IS NOT NULL AND rn.referenceNo IS NOT NULL THEN 'QRR'
+             WHEN REPLACE(COALESCE(orgbpb_invoice.iban, orgbpb.iban, ''), ' ', '') IS NOT NULL AND REPLACE(orgbpb.sepa_creditoridentifier, ' ', '') IS NOT NULL
+                                                                                                                                 THEN 'SCOR'
+                                                                                                                                 ELSE 'NON'
+         END) || E'\n' || --ReferenceType
+        (CASE
+             WHEN REPLACE(COALESCE(orgbpb_invoice.qr_iban, orgbpb.iban, ''), ' ', '') IS NOT NULL AND rn.referenceNo IS NOT NULL THEN rn.ReferenceNo
+             WHEN REPLACE(COALESCE(orgbpb_invoice.iban, orgbpb.iban, ''), ' ', '') IS NOT NULL AND REPLACE(orgbpb.sepa_creditoridentifier, ' ', '') IS NOT NULL
+                                                                                                                                 THEN 'RF' || orgbpb.sepa_creditoridentifier
+                                                                                                                                 ELSE ''
+         END) || E'\n' || --Reference
+        de_metas_endcustomer_fresh_reports.format_invoice_description_for_qr_code(COALESCE(i.description, '')) || E'\n' ||--Unstructured message
```

Comment:
just for my understanding: this is the actual change, right?
besides that, all the changes from this function are formatting, right?

Reply:
@teosarca yes, right!